libkbfs/folder_branch_ops.go

```diff
@@ -3751,17 +3989,25 @@ func (fbo *folderBranchOps) UnregisterFromChanges(obs Observer) error {
 	return nil
 }
 
-// notifyBatchLocked sends out a notification for the most recent op
-// in md.
+// notifyBatchLocked sends out a notification for all the ops in md.
 func (fbo *folderBranchOps) notifyBatchLocked(
 	ctx context.Context, lState *lockState, md ImmutableRootMetadata,
 	afterUpdateFn func() error) error {
 	fbo.headLock.AssertLocked(lState)
 
-	lastOp := md.data.Changes.Ops[len(md.data.Changes.Ops)-1]
-	err := fbo.notifyOneOpLocked(ctx, lState, lastOp, md, false, afterUpdateFn)
-	if err != nil {
-		return err
+	for i, op := range md.data.Changes.Ops {
```

Comment:
This is a bit more confusing than it needs to be, even with the comment. I think it's clearer if you split it out into len = 0, 1, and >1 cases. What do you think?